rbi/stdlib/psych.rbi

```diff
@@ -271,128 +271,99 @@ module Psych
-  # be used in the exception message if any exception is raised while parsing.
-  # If `yaml` is empty, it returns the specified `fallback` return value, which
-  # defaults to `false`.
+  # Load +yaml+ in to a Ruby data structure.  If multiple documents are
+  # provided, the object contained in the first document will be returned.
+  # +filename+ will be used in the exception message if any exception
+  # is raised while parsing.  If +yaml+ is empty, it returns
+  # the specified +fallback+ return value, which defaults to +false+.
   #
-  # Raises a
-  # [`Psych::SyntaxError`](https://docs.ruby-lang.org/en/2.7.0/Psych/SyntaxError.html)
-  # when a [`YAML`](https://docs.ruby-lang.org/en/2.7.0/YAML.html) syntax error
-  # is detected.
+  # Raises a Psych::SyntaxError when a YAML syntax error is detected.
   #
   # Example:
   #
-  # ```ruby
-  # Psych.load("--- a")             # => 'a'
-  # Psych.load("---\n - a\n - b")   # => ['a', 'b']
+  #   Psych.load("--- a")             # => 'a'
+  #   Psych.load("---\n - a\n - b")   # => ['a', 'b']
   #
-  # begin
-  #   Psych.load("--- `", filename: "file.txt")
-  # rescue Psych::SyntaxError => ex
-  #   ex.file    # => 'file.txt'
-  #   ex.message # => "(file.txt): found character that cannot start any token"
-  # end
-  # ```
+  #   begin
+  #     Psych.load("--- `", filename: "file.txt")
+  #   rescue Psych::SyntaxError => ex
+  #     ex.file    # => 'file.txt'
+  #     ex.message # => "(file.txt): found character that cannot start any token"
+  #   end
   #
-  # When the optional `symbolize_names` keyword argument is set to a true value,
-  # returns symbols for keys in
-  # [`Hash`](https://docs.ruby-lang.org/en/2.7.0/Hash.html) objects (default:
-  # strings).
+  # When the optional +symbolize_names+ keyword argument is set to a
+  # true value, returns symbols for keys in Hash objects (default: strings).
   #
-  # ```ruby
-  # Psych.load("---\n foo: bar")                         # => {"foo"=>"bar"}
-  # Psych.load("---\n foo: bar", symbolize_names: true)  # => {:foo=>"bar"}
-  # ```
+  #   Psych.load("---\n foo: bar")                         # => {"foo"=>"bar"}
+  #   Psych.load("---\n foo: bar", symbolize_names: true)  # => {:foo=>"bar"}
+  #
+  # Raises a TypeError when `yaml` parameter is NilClass.  This method is
+  # similar to `safe_load` except that `Symbol` objects are allowed by default.
   #
-  # Raises a [`TypeError`](https://docs.ruby-lang.org/en/2.7.0/TypeError.html)
-  # when `yaml` parameter is
-  # [`NilClass`](https://docs.ruby-lang.org/en/2.7.0/NilClass.html)
-  #
-  # NOTE: This method \*should not\* be used to parse untrusted documents, such
-  # as [`YAML`](https://docs.ruby-lang.org/en/2.7.0/YAML.html) documents that
-  # are supplied via user input. Instead, please use the
-  # [`safe_load`](https://docs.ruby-lang.org/en/2.7.0/Psych.html#method-c-safe_load)
-  # method.
   sig do
     params(
       yaml: T.any(String, StringIO, IO),
-      legacy_filename: Object,
```

Comment:
For backwards compatibility, I think this should stay in place and not be removed.

Reply:
fixed